design-book/src/design-constraints.md

```diff
@@ -0,0 +1,53 @@
+# Design Constraints
+
+To help ensure that we're building an editor that's both useful to end users and feasible to maintain,
+we've developed a set of design constraints and desiderata. Please try to follow them when making and reviewing changes!
+As we're developing, initial prototypes may bend or break the rules: just be mindful of how we can move towards the final desirable end goal.
+
+Below each bullet point, motivation is given.
+
+## Must
```

Comment:
Maybe it's worth defining which target platforms are necessary to support? I assume Windows 11, Linux Wayland, Linux X11, macOS ARM, but idk about macOS intel since we have a ton of Bevy bugs there anyways